sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -3375,6 +3375,12 @@ inline pi_result piextUSMEnqueueMemAdvise(pi_queue Queue, const void *Ptr,
   if (Advice & PI_MEM_ADVICE_CUDA_UNSET_ACCESSED_BY_HOST) {
     UrAdvice |= UR_USM_ADVICE_FLAG_CLEAR_ACCESSED_BY_HOST;
   }
+  if (Advice & PI_MEM_ADVICE_HIP_SET_COARSE_GRAINED) {

```

Comment:
I'm not sure how `UrAdvice` is used later on (so feel free to disregard this comment), but I'm wondering if it makes sense for "toggle" options (such as switch on/off coarse grain mem advice) to also unset the opposite value. Since the values of toggles are not exclusive (10th and 11th bit flipped on for mem advice) you could end up in a situation when both are set and it'd be down to implementation to decide what to do?

As in, how would the implementation react to this (slightly forced bit of code):
```cpp
q.mem_advise(ptr, sizeof(int), int{PI_MEM_ADVICE_HIP_SET_COARSE_GRAINED});
q.mem_advise(ptr, sizeof(int), int{PI_MEM_ADVICE_HIP_UNSET_COARSE_GRAINED});
```
My worry is that somewhere there is a snippet like:
```cpp
if (UrAdvice & PI_MEM_ADVICE_HIP_SET_COARSE_GRAINED)
  do blah
```
and, if I understand it correctly `do blah` would now get triggered, no?

Reply:
Thanks for raising this. It's been used like this for a while, before these changes. I can see the concern but in the adapters each URAdvice set flag is mapped to a underlying value. Each value is then processed in a loop, preserving the natural set/unset order.

Any other use-case of URAdvice in our adapters that is outside of the usm madvise API might suffer from the flakiness of this design. If the need occurs in future, we may follow-up to resolve this.